airbyte-webapp/src/views/Connector/ServiceForm/ServiceForm.tsx

```diff
@@ -83,9 +84,40 @@ const PatchInitialValuesWithWidgetConfig: React.FC<{ schema: JSONSchema7 }> = ({
   return null;
 };
 
+/**
+ * A component that will observe whenever the serviceType (selected connector)
+ * changes and set the name of the connector to match the connector definition name.
+ */
+const SetDefaultName: React.FC<{
+  availableConnectors: ConnectorDefinition[];
+}> = ({ availableConnectors }) => {
+  const { setFieldValue, setFieldTouched } = useFormikContext();
+  const [serviceTypeField] = useField("serviceType");
+
+  useEffect(() => {
+    const name =
+      availableConnectors.find(
+        (service) => Connector.id(service) === serviceTypeField.value
+      )?.name ?? "";
+    // Formik has an issue, that prevents us from setting a field value directly in code here
+    // It won't change the value at all, unless we push it one execution slot further with setTimeout
+    setTimeout(() => {
```

Comment:
It's a bit strange to hear, that it requires `setTimeout`. What is the problem exactly? We are calling `setFieldTouched` in `usePatchFormik` and it works as expected.

Am I missing something? Is it just not triggered?

Reply:
It's simply not switching the field. Here is a [Codesandbox](https://codesandbox.io/embed/formik-setfieldvalue-not-working-d3fyk) that demonstrates that failure in a minimap reproducible way. It should set the mail to `test` but instead it doesn't do anything (unless you pack it into a `setTimeout`). That example wasn't built by me, but someone else who found the same issue already.

I also validated that nothing is resetting the value or any reeinitialization of the form happens after we call this, so this seems to be an issue in Formik (I assume only triggering when it's running in the same execution slot as the form initialization possibility). `setFieldTouch` (which I also initially used here, but removed it) actually does not have the issue. It's really just the `setValue`/`setFieldValue` calls that are having that issue. I'll still file an issue in the Formik repo for that later.

Reply:
And thanks for mentioning `setFIeldTouched`, I just noticed I unnecessarily have this as a dependency here :)